### PR TITLE
Improve the gRPC CLI help description with proto path details

### PR DESCRIPTION
**Pull request by BuddhiWathsala**
## Purpose

Fixes https://github.com/ballerina-platform/ballerina-standard-library/issues/2588

## Approach
- Add proto path details
- Add directory support information to the input entry
- Remove proxy related information

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: cli/ballerina-cli/src/main/resources/cli-help/ballerina-grpc.help [hunk 1]
```diff
@@ -23,9 +23,11 @@ OPTIONS
 
        --mode client | service | proxy
```
Comment (dilanSachi) [inline cli/ballerina-cli/src/main/resources/cli-help/ballerina-grpc.help:24]:
```suggestion
       --mode client | service
```

File: cli/ballerina-cli/src/main/resources/cli-help/ballerina-grpc.help [hunk 2]
```diff
@@ -12,7 +12,7 @@ DESCRIPTION
 
 OPTIONS
        --input <proto-file-path>
-           Path of the input `.proto` file.
+           Input .proto file or a directory containing multiple .proto files.
```
Comment (daneshk) [inline cli/ballerina-cli/src/main/resources/cli-help/ballerina-grpc.help:15]:
```suggestion
           Path to a .proto file or a directory containing multiple .proto files.
```

File: cli/ballerina-cli/src/main/resources/cli-help/ballerina-grpc.help [hunk 3]
```diff
@@ -23,9 +23,11 @@ OPTIONS
 
        --mode client | service | proxy
            Set the `client` or `service` mode to generate sample code.
-           Set the `proxy` mode to generate a gateway proxy.
            If not specified, only the stub file is generated.
 
+       --proto_path
```
Comment (daneshk) [inline cli/ballerina-cli/src/main/resources/cli-help/ballerina-grpc.help:28]:
```suggestion
       --proto_path <path>
```

File: cli/ballerina-cli/src/main/resources/cli-help/ballerina-grpc.help [hunk 4]
```diff
@@ -12,7 +12,7 @@ DESCRIPTION
 
 OPTIONS
        --input <proto-file-path>
```
Comment (daneshk) [inline cli/ballerina-cli/src/main/resources/cli-help/ballerina-grpc.help:14]:
```suggestion
       --input  <path>
```

Comment (BuddhiWathsala) [reply cli/ballerina-cli/src/main/resources/cli-help/ballerina-grpc.help]: Fixed in https://github.com/ballerina-platform/ballerina-lang/pull/34686/commits/fdbf3d28db757767a7ef880cc166507ea51caaa7

File: cli/ballerina-cli/src/main/resources/cli-help/ballerina-grpc.help (hunk 3 above)
Comment (BuddhiWathsala) [reply cli/ballerina-cli/src/main/resources/cli-help/ballerina-grpc.help]: Fixed in https://github.com/ballerina-platform/ballerina-lang/pull/34686/commits/fdbf3d28db757767a7ef880cc166507ea51caaa7

File: cli/ballerina-cli/src/main/resources/cli-help/ballerina-grpc.help (hunk 2 above)
Comment (BuddhiWathsala) [reply cli/ballerina-cli/src/main/resources/cli-help/ballerina-grpc.help]: Fixed in https://github.com/ballerina-platform/ballerina-lang/pull/34686/commits/fdbf3d28db757767a7ef880cc166507ea51caaa7

File: cli/ballerina-cli/src/main/resources/cli-help/ballerina-grpc.help (hunk 1 above)
Comment (BuddhiWathsala) [reply cli/ballerina-cli/src/main/resources/cli-help/ballerina-grpc.help]: Fixed in https://github.com/ballerina-platform/ballerina-lang/pull/34686/commits/fdbf3d28db757767a7ef880cc166507ea51caaa7

File: cli/ballerina-cli/src/main/resources/cli-help/ballerina-grpc.help [hunk 5]
```diff
@@ -4,15 +4,15 @@ NAME
 
 SYNOPSIS
        bal grpc [--input <proto-file-path>] [--output <path>]
-                        [--mode] (client | service | proxy)
+                        [--mode] (client | service)
```
Comment (daneshk) [inline cli/ballerina-cli/src/main/resources/cli-help/ballerina-grpc.help:7]:
```suggestion
                        [--mode] (client | service)
                        [--proto_path] <path>
```

File: cli/ballerina-cli/src/main/resources/cli-help/ballerina-grpc.help [hunk 6]
```diff
@@ -4,15 +4,15 @@ NAME
 
 SYNOPSIS
        bal grpc [--input <proto-file-path>] [--output <path>]
```
Comment (daneshk) [inline cli/ballerina-cli/src/main/resources/cli-help/ballerina-grpc.help:6]:
```suggestion
       bal grpc [--input] <proto-file-path> [--output] <path>
```

**Review by MadhukaHarith92 — DISMISSED** [review]
LGTM

File: cli/ballerina-cli/src/main/resources/cli-help/ballerina-grpc.help (hunk 6 above)
Comment (BuddhiWathsala) [reply cli/ballerina-cli/src/main/resources/cli-help/ballerina-grpc.help]: Fixed in https://github.com/ballerina-platform/ballerina-lang/pull/34686/commits/7748dedec94cfcd8b78279971effbc969478f5d5

File: cli/ballerina-cli/src/main/resources/cli-help/ballerina-grpc.help (hunk 5 above)
Comment (BuddhiWathsala) [reply cli/ballerina-cli/src/main/resources/cli-help/ballerina-grpc.help:7]: Fixed in https://github.com/ballerina-platform/ballerina-lang/pull/34686/commits/7748dedec94cfcd8b78279971effbc969478f5d5

**Review by daneshk — DISMISSED** [review]
LGTM

**Review by dilanSachi — DISMISSED** [review]
LGTM

File: cli/ballerina-cli/src/main/resources/cli-help/ballerina-grpc.help [hunk 7]
```diff
@@ -3,16 +3,17 @@ NAME
                         definition
 
 SYNOPSIS
-       bal grpc [--input <proto-file-path>] [--output <path>]
-                        [--mode] (client | service | proxy)
+       bal grpc [--input] <proto-file-path> [--output] <path>
```
Comment (sameerajayasoma) [inline cli/ballerina-cli/src/main/resources/cli-help/ballerina-grpc.help:6]: Is the `--input` optional? In that case, it should be `[--input <proto-file-path>]`

File: cli/ballerina-cli/src/main/resources/cli-help/ballerina-grpc.help [hunk 8]
```diff
@@ -3,16 +3,17 @@ NAME
...
 
 DESCRIPTION
        Generate the Ballerina gRPC client/service sources for a given
        gRPC protocol buffer (Protobuf) definition.
 
 OPTIONS
-       --input <proto-file-path>
-           Path of the input `.proto` file.
+       --input <path>
+           Path to a .proto file or a directory containing multiple .proto files.
```
Comment (sameerajayasoma) [inline cli/ballerina-cli/src/main/resources/cli-help/ballerina-grpc.help:16]:
```suggestion
           Path to a '.proto' file or a directory containing multiple '.proto' files.
```

File: cli/ballerina-cli/src/main/resources/cli-help/ballerina-grpc.help [hunk 9]
```diff
@@ -21,11 +22,13 @@ OPTIONS
            definition. If a package is not specified, the output will be written
            to a 'temp' directory in the current location.
 
-       --mode client | service | proxy
+       --mode client | service
            Set the `client` or `service` mode to generate sample code.
-           Set the `proxy` mode to generate a gateway proxy.
            If not specified, only the stub file is generated.
 
+       --proto_path <path>
```
Comment (sameerajayasoma) [inline cli/ballerina-cli/src/main/resources/cli-help/ballerina-grpc.help:29]:
```suggestion
       --proto-path <path>
```

Comment (sameerajayasoma) [reply cli/ballerina-cli/src/main/resources/cli-help/ballerina-grpc.help:29]: Option names use dashes

File: cli/ballerina-cli/src/main/resources/cli-help/ballerina-grpc.help [hunk 10]
```diff
@@ -3,16 +3,17 @@ NAME
                         definition
 
 SYNOPSIS
-       bal grpc [--input <proto-file-path>] [--output <path>]
-                        [--mode] (client | service | proxy)
+       bal grpc [--input] <proto-file-path> [--output] <path>
+                        [--mode] (client | service)
+                        [--proto_path] <path>
```
Comment (sameerajayasoma) [inline cli/ballerina-cli/src/main/resources/cli-help/ballerina-grpc.help:8]:
```suggestion
                        [--proto-path] <path>
```

File: cli/ballerina-cli/src/main/resources/cli-help/ballerina-grpc.help (hunk 7 above)
Comment (sameerajayasoma) [inline cli/ballerina-cli/src/main/resources/cli-help/ballerina-grpc.help:6]:
```suggestion
       bal grpc --input <proto-file-path> --output <path>
```

File: cli/ballerina-cli/src/main/resources/cli-help/ballerina-grpc.help (hunk 9 above)
Comment (daneshk) [reply cli/ballerina-cli/src/main/resources/cli-help/ballerina-grpc.help:29]: we referred to the protoc command and use the same option key. https://developers.google.com/protocol-buffers/docs/proto3#generating. We will change it to dash

File: cli/ballerina-cli/src/main/resources/cli-help/ballerina-grpc.help [hunk 11]
```diff
@@ -3,16 +3,17 @@ NAME
                         definition
 
 SYNOPSIS
-       bal grpc [--input <proto-file-path>] [--output <path>]
-                        [--mode] (client | service | proxy)
+       bal grpc [--input] <proto-file-path> [--output] <path>
+                        [--mode] (client | service)
```
Comment (sameerajayasoma) [inline cli/ballerina-cli/src/main/resources/cli-help/ballerina-grpc.help:7]:
```suggestion
                        [--mode mode ]
```

Comment (sameerajayasoma) [reply cli/ballerina-cli/src/main/resources/cli-help/ballerina-grpc.help:7]: You can specify the mode values in the option description

File: cli/ballerina-cli/src/main/resources/cli-help/ballerina-grpc.help [hunk 12]
```diff
@@ -21,11 +22,13 @@ OPTIONS
            definition. If a package is not specified, the output will be written
            to a 'temp' directory in the current location.
 
-       --mode client | service | proxy
+       --mode client | service
            Set the `client` or `service` mode to generate sample code.
```
Comment (sameerajayasoma) [inline cli/ballerina-cli/src/main/resources/cli-help/ballerina-grpc.help:26]:
```suggestion
           Set the 'client' or 'service' mode to generate sample code.
```

File: cli/ballerina-cli/src/main/resources/cli-help/ballerina-grpc.help [hunk 13]
```diff
@@ -21,11 +22,13 @@ OPTIONS
            definition. If a package is not specified, the output will be written
            to a 'temp' directory in the current location.
 
-       --mode client | service | proxy
+       --mode client | service
```
Comment (sameerajayasoma) [inline cli/ballerina-cli/src/main/resources/cli-help/ballerina-grpc.help:25]:
```suggestion
       --mode mode
```

File: cli/ballerina-cli/src/main/resources/cli-help/ballerina-grpc.help (hunk 7 above)
Comment (BuddhiWathsala) [reply cli/ballerina-cli/src/main/resources/cli-help/ballerina-grpc.help]: Fixed in https://github.com/ballerina-platform/ballerina-lang/pull/34686/commits/ddaea2a9d5eecb9756bcc418a37a99ee8672aeda

Comment (BuddhiWathsala) [reply cli/ballerina-cli/src/main/resources/cli-help/ballerina-grpc.help]: Fixed in https://github.com/ballerina-platform/ballerina-lang/pull/34686/commits/ddaea2a9d5eecb9756bcc418a37a99ee8672aeda

File: cli/ballerina-cli/src/main/resources/cli-help/ballerina-grpc.help (hunk 12 above)
Comment (BuddhiWathsala) [reply cli/ballerina-cli/src/main/resources/cli-help/ballerina-grpc.help]: Fixed in https://github.com/ballerina-platform/ballerina-lang/pull/34686/commits/0ba70265ce04e785ee7394104e66b7355e09ef49

File: cli/ballerina-cli/src/main/resources/cli-help/ballerina-grpc.help (hunk 8 above)
Comment (BuddhiWathsala) [reply cli/ballerina-cli/src/main/resources/cli-help/ballerina-grpc.help]: Fixed in https://github.com/ballerina-platform/ballerina-lang/pull/34686/commits/0ba70265ce04e785ee7394104e66b7355e09ef49

File: cli/ballerina-cli/src/main/resources/cli-help/ballerina-grpc.help (hunk 13 above)
Comment (BuddhiWathsala) [reply cli/ballerina-cli/src/main/resources/cli-help/ballerina-grpc.help]: Fixed in https://github.com/ballerina-platform/ballerina-lang/pull/34686/commits/c041ce5fcdfe872fc4378580d5d0b975ecbde922

File: cli/ballerina-cli/src/main/resources/cli-help/ballerina-grpc.help (hunk 11 above)
Comment (BuddhiWathsala) [reply cli/ballerina-cli/src/main/resources/cli-help/ballerina-grpc.help]: Fixed in https://github.com/ballerina-platform/ballerina-lang/pull/34686/commits/c041ce5fcdfe872fc4378580d5d0b975ecbde922

File: cli/ballerina-cli/src/main/resources/cli-help/ballerina-grpc.help (hunk 9 above)
Comment (sameerajayasoma) [reply cli/ballerina-cli/src/main/resources/cli-help/ballerina-grpc.help:29]: POSIX command generally uses dashes, hence the suggestion. :)

File: cli/ballerina-cli/src/main/resources/cli-help/ballerina-grpc.help (hunk 10 above)
Comment (BuddhiWathsala) [reply cli/ballerina-cli/src/main/resources/cli-help/ballerina-grpc.help]: Added a task https://github.com/ballerina-platform/ballerina-standard-library/issues/2590

File: cli/ballerina-cli/src/main/resources/cli-help/ballerina-grpc.help (hunk 9 above)
Comment (BuddhiWathsala) [reply cli/ballerina-cli/src/main/resources/cli-help/ballerina-grpc.help:29]: Added a task https://github.com/ballerina-platform/ballerina-standard-library/issues/2590

**Review by daneshk — APPROVED** [review]
LGTM

**Comment by BuddhiWathsala** [comment]
@sameerajayasoma, could you please review the changes?.

**Review by sameerajayasoma — APPROVED** [review]

**Review by pubudu91 — APPROVED** [review]

**Pull request merged**